lib/components/search.jsx

```diff
@@ -0,0 +1,202 @@
+    super(props)
+    const { terms } = this.props
+    this.state = {
+      isInstalling: false,
+      isLoading: null,
+      matches: [],
+      selectedPackage: null,
+      terms: terms || ''
+    }
+    this.onChangeTerms = this.onChangeTerms.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+    this.onSelectPackage = this.onSelectPackage.bind(this)
+  }
+
+  componentDidMount () {
+    const { terms } = this.props
+    if (terms) {
+      this.search(terms)
+    }
+  }
+
+  render () {
+    const {
+      isInstalling,
+      isLoading,
+      matches,
+      selectedPackage,
+      terms
+    } = this.state
+
+    return <div>
+      <SearchInput
+        terms={terms}
+        onChange={this.onChangeTerms}
+        onSubmit={this.onSubmit} />
+      <SearchResults
+        isLoading={isLoading}
+        terms={terms}
+        matches={matches}
+        onSelect={this.onSelectPackage} />
+      { isInstalling && selectedPackage
+        ? <InstallingPackage isInstalling={isInstalling} pkg={selectedPackage} />
+        : null
+      }
+    </div>
+  }
+
+  async search (terms) {
+    try {
+      this.setState({
+        isLoading: true,
+        matches: []
+      })
+      const matches = await libnpm.search(terms, {
+        detailed: true,
+        limit: MAX_RESULTS
+      })
+      this.setState({
+        isLoading: false,
+        matches
+      })
+    } catch (err) {
+      // @todo Show error message
+      this.setState({
+        isLoading: false,
+        matches: []
+      })
+    }
+  }
+
+  async install (pkg) {
+    try {
+      this.setState({
+        isInstalling: true,
+        selectedPackage: pkg
+      })
+      // @todo Implement real install
+      await new Promise((resolve, reject) => setTimeout(resolve, 1000))
+      this.setState({ isInstalling: false })
+      this.props.onExit()
+    } catch (err) {
+      // @todo Show error message
+      this.setState({ isInstalling: false })
+    }
+  }
+
+  onChangeTerms (terms) {
+    this.setState({ terms })
+  }
+
+  onSubmit (terms) {
+    this.search(terms)
+  }
+
+  onSelectPackage (pkg) {
+    this.install(pkg)
+  }
+}
+
+const SearchInput = ({ onChange, onSubmit, terms }) => {
+  return <div>
+    <Color grey bold>Find a package: </Color>
+    <TextInput value={terms} onChange={onChange} onSubmit={onSubmit} />
+  </div>
+}
+
+const SearchResults = ({ isLoading, matches, onSelect, terms }) => {
+  if (isLoading) {
+    return <div><Spinner /> <Color green>Searching...</Color></div>
+  }
+
+  if (isLoading === false && terms && (!matches || !matches.length)) {
+    return <Color grey>No matches found</Color>
+  }
+
+  if (isLoading === false && terms && matches && matches.length) {
+    return <div>
+      <Color grey bold>Results:</Color><br />
+      <PackageSelector matches={matches} onSelect={onSelect} />
+    </div>
+  }
+
+  return null
+}
+
+const PackageSelectIndicator = ({ isSelected }) => {
+  if (!isSelected) {
+    return ' ';
```

Comment:
Shouldn't this be two spaces so that it is the same length as when it is selected?
```suggestion
    return '  ';
```

Reply:
I just tested this and... it doesn't work? If I change it to a single `#` the indentation is one char (no spaces), as expected. However, when I keep it as a single space it shows two spaces and when I add the second it shows three... There are some weird things going on with spacing here and there in either ink or jsx.

Reply:
Huh. That's really interesting.

Reply:
I guess it tried to simulate html in that aspect lol